File: server/src/main/java/com/noahgeren/trailangel/domain/Hike.java
package com.noahgeren.trailangel.domain;

import java.time.LocalDateTime;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

import com.fasterxml.jackson.annotation.JsonIgnore;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class Hike {
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;
	
	private String trailName;
	
	private Double latitude, longitude;
	
	private Double length;
	
	private LocalDateTime startTime;
	
	private LocalDateTime endTime;
	
	private String user;

}

File: client/app/src/main/java/com/noahgeren/trailangel/models/Hike.kt
package com.noahgeren.trailangel.models

import android.annotation.SuppressLint
import android.os.Build
import androidx.annotation.RequiresApi
import androidx.room.Entity
import androidx.room.PrimaryKey
import java.io.Serializable
import java.time.LocalDateTime

@Entity
@RequiresApi(Build.VERSION_CODES.O)
data class Hike (
    @PrimaryKey var id: Int? = null,
    var trailName: String = "",
    var latitude: Double? = null,
    var longitude: Double? = null,
    var length: Double? = null,
    var startTime: LocalDateTime = LocalDateTime.now(),
    var endTime: LocalDateTime = LocalDateTime.now(),
    var user: String? = null
): Serializable
